### ISSUE TITLE
Bug: `panic: runtime error: index out of range [2] with length 2` - building schema

### ISSUE THREAD

**Issue by TristanSpeakEasy** [pb33f/libopenapi#43]
I get the below panic when parsing some OpenAPI documents:

```
panic: runtime error: index out of range [2] with length 2

goroutine 1 [running]:
github.com/pb33f/libopenapi/utils.FindKeyNodeTop({0xc0004191a0, 0x9}, {0xc0002cf430, 0x2, 0x84ccf0?})
        /home/trist/go/pkg/mod/github.com/pb33f/libopenapi@v0.4.0/utils/utils.go:209 +0xe7
github.com/pb33f/libopenapi/datamodel/low.BuildModel(0xc000309c20, {0x7350c0?, 0xc0006eed80?})
        /home/trist/go/pkg/mod/github.com/pb33f/libopenapi@v0.4.0/datamodel/low/model_builder.go:43 +0x4b0
github.com/pb33f/libopenapi/datamodel/low/base.(*Schema).Build(0xc000462800, 0x75f500?, 0xc000100c00)
        /home/trist/go/pkg/mod/github.com/pb33f/libopenapi@v0.4.0/datamodel/low/base/schema.go:629 +0x17a5
github.com/pb33f/libopenapi/datamodel/low/base.(*SchemaProxy).Schema(0xc000636ff0)
        /home/trist/go/pkg/mod/github.com/pb33f/libopenapi@v0.4.0/datamodel/low/base/schema_proxy.go:84 +0x77
github.com/pb33f/libopenapi/datamodel/high/base.(*SchemaProxy).Schema(0xc000287f80)
        /home/trist/go/pkg/mod/github.com/pb33f/libopenapi@v0.4.0/datamodel/high/base/schema_proxy.go:58 +0x2b
github.com/pb33f/libopenapi/datamodel/high/base.(*SchemaProxy).BuildSchema(0xc000287f80)
        /home/trist/go/pkg/mod/github.com/pb33f/libopenapi@v0.4.0/datamodel/high/base/schema_proxy.go:70 +0x1e
main.main()
        /home/trist/workspace/scratch/main.go:48 +0x2c8
exit status 2
```

here is some example code that reproduces it:
```go
package main

import (
	"io"
	"net/http"

	"github.com/pb33f/libopenapi"
)

func main() {
	res, err := http.Get("https://raw.githubusercontent.com/speakeasy-api/openapi-directory/main/APIs/amazonaws.com/sdb/2009-04-15/openapi.yaml")
	if err != nil {
		panic(err)
	}
	defer res.Body.Close()

	data, err := io.ReadAll(res.Body)
	if err != nil {
		panic(err)
	}

	doc, err := libopenapi.NewDocument(data)
	if err != nil {
		panic(err)
	}

	model, errs := doc.BuildV3Model()
	if len(errs) > 0 {
		panic(errs)
	}

	for _, item := range model.Model.Paths.PathItems {
		for _, op := range item.GetOperations() {
			for _, param := range op.Parameters {
				schema, err := param.Schema.BuildSchema()
				if err != nil {
					panic(err)
				}

				if schema.Items != nil && schema.Items.IsA() {
					itemSchema, err := schema.Items.A.BuildSchema()
					if err != nil {
						panic(err)
					}

					if itemSchema.AllOf != nil {
						for _, s := range itemSchema.AllOf {
							_, err := s.BuildSchema() // PANICS HERE
							if err != nil {
								panic(err)
							}
						}
					}
				}
			}
		}
	}
}
```

**Comment by daveshanley** [comment]
Interesting, I'll look into it.

**Comment by daveshanley** [comment]
This should be resolved in `v0.4.3`.

**Comment by TristanSpeakEasy** [comment]
can confirm fixed
